src/sparse_int_set.jl

```diff
@@ -0,0 +1,265 @@
+import Base: @propagate_inbounds, zip
+
+const INT_PER_PAGE = div(ccall(:jl_getpagesize, Clong, ()), sizeof(Int))
+const NULL_INT_PAGE = Vector{Int}()
+
+#TODO: Batch creation and allocation
+mutable struct SparseIntSet
+    packed ::Vector{Int}
+    reverse::Vector{Vector{Int}}
+    counters::Vector{Int}  # counts the number of real elements in each page of reverse. 
+end
+
+SparseIntSet() = SparseIntSet(Int[], Vector{Int}[], Int[])
+
+SparseIntSet(indices) = union!(SparseIntSet(), indices)
+
+eltype(::Type{SparseIntSet}) = Int
+
+empty(::SparseIntSet) = SparseIntSet()
+
+function empty!(s::SparseIntSet)
+    empty!(s.packed)
+    empty!(s.reverse)
+    empty!(s.counters)
+    return s
+end
+
+isempty(s::SparseIntSet) = isempty(s.packed)
+
+copy(s::SparseIntSet) = copy!(SparseIntSet(), s)
+
+function copy!(to::SparseIntSet, from::SparseIntSet)
+    to.packed = copy(from.packed)
+    to.reverse = deepcopy(from.reverse)
+    to.counters = copy(from.counters)
+    return to
+end
+
+function pageid_offset(s::SparseIntSet, i)
+    pageid = div(i - 1, INT_PER_PAGE) + 1
+    return pageid, (i - 1) & (INT_PER_PAGE - 1) + 1
+end
+
+function in(i, s::SparseIntSet)
+    pageid, offset = pageid_offset(s, i)
+    isassigned(s.reverse, pageid) && @inbounds s.reverse[pageid][offset] != 0
```

Comment:
```suggestion
    page = @inbounds s.reverse[pageid]
    return page !=== NULL_INT_PAGE && @inbounds page[offset] != 0
```

Reply:
Need a test that catches this case

Reply:
Wait a second, won't this potentially segfault if the pageid is larger than the amount of pages?

Reply:
I added a length check, seems to be fine now.